src/base64.cr

```diff
@@ -83,6 +83,31 @@ module Base64
     end
   end
 
+  private def encode_with_new_lines(io : IO, &)
+    inc = 0
+    # Base64 operates in 3-byte segments, so we use a buffer size that is a
+    # multiple of 3.
+    buffer = StaticArray(UInt8, 8193).new { 0u8 }
+    bytes = Bytes.new(buffer.to_unsafe, buffer.size)
+
+    loop do
+      bytes_read = io.read(bytes)
+      break if bytes_read == 0
+
+      to_base64(bytes[0, bytes_read], CHARS_STD, pad: bytes_read < bytes.bytesize) do |byte|
```

Comment:
My first implementation had `true` in there and it added padding internally. That's the reason I added the inequality expression.

Can you provide any failing tests for cases you've found where this doesn't work?

Reply:
> Then disabling the explicit padding only hides that the code doesn't work.

You mean it doesn't work for this one very specific and pathological case? Saying it doesn't work, unqualified, is incorrect. If it were true, it wouldn't have passed the specs I included.

Reply:
`IO#read` is only guaranteed to read at least 1 byte. You need to account for that and cannot assume `bytes` is filled.
The specs work because they use `IO::Memory` which has all data available in memory and thus can always fill the read buffer to the brim. If you read from an IO that needs to wait for data to become available, it may give you much smaller chunks than `bytes.bytesize` (even a buffered IO).
`io_spec.cr` and other specs have special IO implementations that simulate this behaviour, like the `BadButValidIO` example.

Reply:
I didn't object to handling that case. I was responding to an unqualified "the code doesn't work".